File: app/src/main/java/com/relic/data/gateway/UserGatewayImpl.java
package com.relic.data.gateway;

import android.content.Context;
import android.os.AsyncTask;
import android.util.Log;

import com.relic.R;
import com.relic.network.NetworkRequestManager;
import com.relic.network.request.RelicOAuthRequest;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class UserGatewayImpl implements UserGateway {
    public static String TAG = "USER_GATEWAY";
    private final String ENDPOINT = "https://oauth.reddit.com/";

    private NetworkRequestManager requestManager;

    public UserGatewayImpl(Context context, NetworkRequestManager networkRequestManager) {
        requestManager = networkRequestManager;
    }

    public void getUser(String username) {
        String endpoint = ENDPOINT + "/user/" + username + "/about";
        requestManager.processRequest(new RelicOAuthRequest(
            RelicOAuthRequest.GET,
            endpoint,
            response -> {
                parseUser(response);
            },
            error -> {
                Log.d(TAG, "Error getting user overview");
            }
        ));
    }

    private void parseUser(String response) {
        Log.d(TAG, response);
        JSONParser parser = new JSONParser();

        try {
            JSONObject full = (JSONObject) ((JSONObject) parser.parse(response)).get("data");

            Log.d(TAG, full.keySet().toString());
        } catch (ParseException e) {
            Log.d(TAG, "Error parsing the response");
        }
    }


    private static class another extends AsyncTask {
        @Override
        protected Object doInBackground(Object[] objects) {
            return null;
        }
    }


}

File: app/src/main/java/com/relic/data/gateway/UserGatewayImpl.kt
package com.relic.data.gateway

import com.relic.data.DomainTransfer
import com.relic.network.NetworkRequestManager
import com.relic.network.request.RelicOAuthRequest
import org.json.simple.JSONObject
import org.json.simple.parser.JSONParser
import org.json.simple.parser.ParseException
import timber.log.Timber
import javax.inject.Inject

class UserGatewayImpl @Inject constructor(
    private val requestManager: NetworkRequestManager
) : UserGateway {
    private val ENDPOINT = "https://oauth.reddit.com/"

    override suspend fun getUser(username: String) {
        val endpoint = "$ENDPOINT/user/$username/about"
        try {
            val response = requestManager.processRequest(RelicOAuthRequest.GET, endpoint)
            parseUser(response)

        } catch (e: Exception) {
            throw DomainTransfer.handleException("refresh token", e) ?: e
        }
    }

    private fun parseUser(response: String) {
        Timber.d(response)
        val parser = JSONParser()

        try {
            val full = (parser.parse(response) as JSONObject)["data"] as JSONObject?

            Timber.d(full!!.keys.toString())
        } catch (e: ParseException) {
            Timber.d("Error parsing the response")
        }

    }

    companion object {
        var TAG = "USER_GATEWAY"
    }
}
